knowledge_repo/app/index.py

```diff
@@ -40,7 +40,7 @@ def update_index():
         kp = kr_dir.pop(post.path)
 
         # Update metadata of post if required
-        if (kp.revision > post.revision or not post.is_published or kp.uuid != post.uuid):
+        if (post.revision is None or kp.revision > post.revision or not post.is_published or kp.uuid != post.uuid):
```

Comment:
@matthewwardrop in python 3 if you try to compare None to an integer, it throws an exception